src/LayerEditor/ui/panels/regions_panel.py

```diff
@@ -59,87 +50,85 @@ def select_regions(self, region_dict):
         :return:
         """
         # assert layer_id in self._selected_regions
+        assert False, "Not refactored"
         self._selected_regions.update(region_dict)
         self.region_changed.emit()
 
     def select_layer(self, layer_id):
         """ Select actual layer tab."""
-        self._selected_layers[self.current_topology_id] = layer_id
+        assert False, "Not refactored"
+        self.block.gui_selected_layers[self.current_topology_id] = layer_id
 
-    def add_region(self, name, dim, color):
+    def add_region(self, name, dim):
```

Comment:
I wonder why the color is not passed to the region?

Reply:
The color doesn't exist until RegionItem is made and gets default color. Done.